slamd/discovery/processing/experiment/experiment_conductor.py

```diff
@@ -40,25 +41,34 @@ def _fit_model_and_predict(cls, exp):
         else:
             raise ValueNotSupportedException(message=f'Invalid model: {exp.model}')
 
-        predictions = {}
-        uncertainties = {}
+        predictions = pd.DataFrame(columns=exp.target_names, index=exp.index_predicted, dtype=np.float)
+        uncertainties = pd.DataFrame(columns=exp.target_names, index=exp.index_predicted, dtype=np.float)
         for target in exp.target_names:
             # Train the model for every target with the corresponding rows and labels
-            training_rows = exp.features_df.loc[exp.label_index].values
-            training_labels = exp.targets_df.loc[exp.label_index, target].values.reshape(-1, 1)
+            index_labelled = exp.targets_df.index[exp.targets_df[target].notnull()]
+            index_unlabelled = exp.targets_df.index[exp.targets_df[target].isnull()]
+
+            training_rows = exp.features_df.loc[index_labelled].values
+            training_labels = exp.targets_df.loc[index_labelled, target].values.reshape(-1, 1)
 
             regressor.fit(training_rows, training_labels)
 
             # Predict the label for the remaining rows
-            rows_to_predict = exp.features_df.loc[exp.nolabel_index].values
+            rows_to_predict = exp.features_df.loc[index_unlabelled].values
             prediction, uncertainty = regressor.predict(rows_to_predict, return_std=True)
 
-            predictions[target] = prediction
-            uncertainties[target] = uncertainty
+            predictions.loc[index_unlabelled, target] = prediction
+            uncertainties.loc[index_unlabelled, target] = uncertainty
+
+            # Determine rows for which the current target is labelled, but others aren't
+            index_only_curr_labelled = exp.index_partially_labelled.intersection(index_labelled)
```

Comment:
are there any edge cases you could think of where the intersection is empty. If yes, do we need to include exception handling here?

Reply:
The exception being empty is the standard case, since it is only non-empty if we have partial labels. Afaik pandas is generally smart enough to work with empty indices; the fact that our tests work shows that we don't need exception handling here.